compiler/rustc_mir_transform/src/cross_crate_inline.rs

```diff
@@ -50,6 +50,16 @@ fn cross_crate_inlinable(tcx: TyCtxt<'_>, def_id: LocalDefId) -> bool {
         _ => {}
     }
 
+    let sig = tcx.fn_sig(def_id).instantiate_identity();
+    for ty in sig.inputs().skip_binder().iter().chain(std::iter::once(&sig.output().skip_binder()))
+    {
+        // FIXME(f16_f128): in order to avoid crashes building `core`, always inline to skip
+        // codegen if the function is not used.
```

Comment:
if you move this down after `let mir`, you can just iterate over `mir.args_iter()` instead of having to call `fn_sig`

Reply:
I think this would require moving `let mir` above the checks against `opts.incremental`, `OptLevel::No`, and `threashold` to avoid hitting a `return false` before we check for the types. Would this wind up with a perf impact since we don't generate mir here in these cases?

Reply:
Hm, I thought having this check so far down would be fine. But maybe not.

Reply:
Since this is a (temporary, right?) hack, I'd rather the diff be small and localized than pretty. What you've written here is good.

Reply:
> (temporary, right?)

Yes please, I don't think we will be able to stabilize these types until LLVM at least doesn't crash on everything T2+.